verilog/tools/kythe/kythe_facts_extractor.cc

```diff
@@ -0,0 +1,153 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <string>
+
+#include "absl/strings/escaping.h"
+#include "absl/strings/substitute.h"
+#include "kythe_facts_extractor.h"
+#include "kythe_schema_constants.h"
+
+namespace verilog {
+namespace kythe {
+
+void ExtractKytheFacts(IndexingFactNode& node) {
+  KytheFactsExtractor kythe_extractor(node);
+  kythe_extractor.Visit(node);
+}
+
+void KytheFactsExtractor::Visit(IndexingFactNode& node) {
+  const auto tag =
+      static_cast<IndexingFactType>(node.Value().GetIndexingFactType());
+  switch (tag) {
+    case IndexingFactType::kFile: {
+      ExtractFileFact(node);
+      break;
+    }
+    case IndexingFactType::kModule: {
+      ExtractModuleFact(node);
+      break;
+    }
+    case IndexingFactType::kModuleInstance: {
+      ExtractModuleInstanceFact(node);
+      break;
+    }
+    case IndexingFactType::kModulePortInput:
+      break;
+    case IndexingFactType::kModulePortOutput:
+      break;
+  }
+
+  const AutoPop p(&facts_tree_context_, &node);
+  for (auto& child : node.Children()) {
+    Visit(child);
+  }
+}
+
+void KytheFactsExtractor::ExtractFileFact(IndexingFactNode& node) {
+  VName file_vname(file_path_, "", "");
+
+  PrintFact(file_vname, kFactNodeKind, kNodeFile);
+  PrintFact(file_vname, kFactText, node.Value().Anchors()[1].Value());
```

Comment:
Explain `[1]`?

Reply:
This is the one containing code_text. 
And for FactsTree I am accessing the nodes directly using indices.

Reply:
@fangism Is this okay?